bsip-0043.md

```diff
@@ -0,0 +1,96 @@
+    BSIP: 0043
+    Title: Market fee sharing
+    Authors: OpenLedgerApp <https://github.com/OpenLedgerApp>
+    Status: Draft
+    Type: Protocol
+    Created: 2018-08-23
+    Discussion: https://github.com/bitshares/bsips/issues/102
+    Worker:
+
+
+# Abstract
+
+When creating a new asset, the asset issuer is the only beneficiary of the market fees in the current implementation. And the only way to affect the asset community growth is the market fee percentage. For example, one can decrease the market fee and it will result in somewhat larger number of trades with this asset. In this way the asset issuer might get a bigger profit during to increasing the trade volume with this asset.
+
+However, there might be another opportunity to promote the asset and stimulate the trading - use native Bitshares referral program. At this time unfortunately an assets owner is not able to share market fees with registrars and referrers to stimulate them to market the asset trading, so we suggest to add this possibility. Furthermore, enabling this feature for MPAs (e.g. bitCNY or bitUSD) can provide additional bounty for Bitshares registrars and referrals which can lead to more traders joining to the ecosystem.
+The workflow may be as following:
+
+An asset issuer defines the market_fee_reward_percent - what percentage of the market fee he wants to share with the registrar in asset options. Registrar defines reward_percent for the referrer for each user(using the already existing BitShares mechanism).
+Market fee reward is accumulated on the user account. The user decides when they want to claim the market fee reward and move it to their active balance. There is another operation created asset_claim_reward_operation. Each user pays network fee to call this operation.
+
+# Motivation
+
+To make promoting BitShares and bringing new users much more attractive to registrars and referrers by sharing UIAs and MPAs market fees with them.
+
+# Rationale
+When 'fill_order_operation' executing at the moment of market fee calculation there will be calculate the reward for the registrar used the parameter market_fee_reward_percent. Then this market_fee_reward_percent of the market fee is split between the registrar and referrer according to the referrer_rewards_percentage, which is set up during the new account registration by registrar (please see the parameters for register_account).
+
+# Specifications
+
+## market_fee_reward_percent asset option
+Percent of market fee that will be paid to buyer's registrar. Set by UIA issuer.
+
+## market_sharing_whitelist asset option
+An optional list of accounts (configurable by the UIA Issuer) could provide increased control over who is eligible to get market rewards.
+
+## New database and wallet api methods
+**get_account_reward(account, asset_symbol)** - Returns information about the reward amount for specific asset.
+**list_account_rewards(account)** - Returns information about the reward amount in various assets
```

Comment:
database api calls should have ``start`` and ``limit`` parameters

Reply:
We think that one user will generally have only a few assets with rewards.  So we don't think pagination is needed in this particular case. On one hand, it's definitely good to follow the common standards. 
On the other hand, in bitshares there are other well-known functions that do not have pagination, for example: 
**list_account_balances(account_id)**

Alternatively, we can change list_account_rewards to **get_account_rewards**. 
Or if following standards is becoming a trend in bitshares, we can add start and limit parameters as you suggested. 
What do you think?

Reply:
Makes sense. I don't particularly care, can leave it as is.